### PR TITLE
Use portable-scala crossProject

### PR DESCRIPTION
**Pull request by BennyHill**
Also added native to the kernel project, just to get the ball rolling there - ref https://github.com/typelevel/cats/issues/1549

There was an issue with issue with the workaround used by simulacrum. However, and with reference to https://github.com/typelevel/cats/issues/2152, I switched it back to `Provided` _and_ then added a post process step to remove from the POM: so hopefully, that is now OK, too.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -39,16 +38,23 @@ lazy val commonSettings = Seq(
...
     "org.typelevel" %%% "machinist" % "0.6.4",
     compilerPlugin("org.scalamacros" %% "paradise" % "2.1.0" cross CrossVersion.patch),
     compilerPlugin("org.spire-math" %% "kind-projector" % "0.9.6")
   ),
+  pomPostProcess := { (node: xml.Node) =>
+    new RuleTransformer(new RewriteRule {
+      override def transform(node: xml.Node): Seq[xml.Node] = node match {
+        case e: xml.Elem
+          if e.label == "dependency" && e.child.exists(child => child.label == "groupId" && child.text == "com.github.mpilquist") => Nil
+        case _ => Seq(node)
```
Comment (kailuowang) [inline build.sbt]: Is this blanket removing every dependencies under "com.github.mpilquist" for the sake of simulacrum. Is this worthwhile than just using `Provided`?

Comment (BennyHill) [reply build.sbt]: it just removes the simulacrum dependency the POM, a known concern, as done in the past elsewhere- eg https://github.com/typelevel/cats-effect/pull/77#issuecomment-340068924 .

But if you would rather a more fine controlled selection, by artifact name, sure-np

Comment (kailuowang) [reply build.sbt]: yeah, I think artifact name would be better, but if it's trickier than group name I'd vote for just leave it as "Provdided", not ideal, but didn't cause any problems in the past.

**Comment by kailuowang** [comment]
Thanks @BennyHill !

**Comment by BennyHill** [comment]
Any updates on this, please? As others are right now, just trying to cross things off my open list.

**Comment by BennyHill** [comment]
Blimey....fast response!! Many thanks!

**Comment by kailuowang** [comment]
Merged into 1.2.x branch which will be merged into master soon (supposedly every commits to 1.2.x should be automatically merge back to master, I just want to avoid creating too many PRs, we are going to deal with this 2 branch situation until we get a 2.13-M4 build released)

**Comment by BennyHill** [comment]
Just fyi, this was originally done for the 2.13.0-m4 work, see https://github.com/typelevel/cats/issues/2267#issuecomment-395489518 . Looks like it is not actually required for that, so can be merged in as you require

**Pull request merged**